docs/src/modules/java/pages/spring-client.adoc

```diff
@@ -0,0 +1,80 @@
+= Consuming Akka Serverless Services using Spring
+
+Akka Serverless services are exposed using REST and GRPC interfaces.
+This guide walks you through the process of consuming these services using Spring.
+
+== Consuming REST endpoints
+
+Create a separate new Springboot project.
+
+For consuming the REST endpoints, we can use various REST client libraries provided via spring or others.
+For instance libraries like RestTemplate, Webflux etc.
+
+In this guide we are looking at steps using a Spring client using Webflux library.
+
+----
+ <dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-webflux</artifactId>
+ </dependency>
+----
+
+Steps:
+
+- Define a Webclient and Configure it to use Akka Serverless service's host and port.
+
+[source,java,indent=0]
+.src/main/java/com/example/client/spring/rest/config/WebClientConfig.java
+----
+include::example$java-valueentity-counter-spring-client/src/main/java/com/example/client/spring/rest/config/WebClientConfig.java[tag=webclientconfig]
+----
+
+- Using the above defined client call a relevant API in Akka Serverless service.
+For e.g. GET, POST etc. endpoints.
+
+[source,java,indent=0]
+.src/main/java/com/example/client/spring/rest/service/CounterServiceImpl.java
+----
+include::example$java-valueentity-counter-spring-client/src/main/java/com/example/client/spring/rest/service/CounterServiceImpl.java[tag=getCurrentCounterCall]
+----
+
+== Consuming GRPC Services
+
+Create a separate new Springboot project.
+
+For Consuming GRPC Services of Akka Serverless service we need to generate grpc client stubs.
+
+You could use protobuf maven plugins for stub creation.
+
+----
+<groupId>org.xolstice.maven.plugins</groupId>
+<artifactId>protobuf-maven-plugin</artifactId>
+----
+
+Following steps walks you through steps for creating
+
+- Get all the proto files from relevant akka serverless service and copy under `src/main/proto` folder.
```

Comment:
This is a good moment to tell users to delete the akka serverless annotation. 
This is annoying, not really user friendly, but we don't have a solution for it for the moment.

To clarify, strictly speaking they don't have to delete it, but if they keep, they need to pass the akka serverless annotation to protoc when compiling. It's annoying to have to instruct users how to configure protoc. We don't even know which plugin they are using and how their build is. 

So it's easier and less inelegant to ask them to delete the annotations.

Reply:
Updated it !